src/controllers/species.controller.ts

```diff
@@ -61,23 +66,40 @@ export class SpeciesController {
     filter?: Filter<Species> & { fields?: { captureCount?: Boolean } },
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
   ): Promise<any[]> {
-    console.log('get /species filter --> ', filter ? filter.where : null);
     // Only include active species
     filter = { ...filter, where: { ...filter?.where, active: true } };
+    const speciesList = await this.speciesRepository.find(filter);
 
+    if (this.requestHasUser(this.request)) {
+      //CheckUserAccess
+      const userRoles: string[] = this.request.user.roleNames;
+      const isAdminUser = userRoles.some((userRole) =>
+        allowedRoles.some((allowedRole) => allowedRole == userRole),
+      );
+      if (isAdminUser) {
+        return this.captureCountForEachSpecies(speciesList);
+      }
+    }
+    return speciesList;
+    // TODO: combine both queries in one query to minimise overhead
+  }
+
+  requestHasUser(request: Request): request is Request & { user: any } {
+    return 'user' in request;
+  }
+
+  async captureCountForEachSpecies(speciesList: Species[]): Promise<any[]> {
     const captureCountStmt =
       `SELECT species_id,COUNT(species_id) as count FROM trees ` +
       `WHERE species_id NOTNULL AND active=true GROUP BY species_id`;
-    // TODO: combine both queries in one query to minimise overhead
-    const [speciesList, captureCounts] = await Promise.all([
-      this.speciesRepository.find(filter),
-      this.treesRepository.execute(captureCountStmt, []),
-    ]);
-
+    const captureCounts = await this.treesRepository.execute(
+      captureCountStmt,
+      [],
+    );
     // Merge the capture counts with the species list
     return speciesList.map((species) => {
       const match = captureCounts.find((item) => {
-        return item.species_id === species.id;
+        return item.species_id == species.id;
```

Comment:
What's the reason for reducing this from `===` to `==`?

Reply:
I have updated it back to `===`. I recheck and update the same.